### ISSUE TITLE
[BUG] TS - pyarrow 21.0.0 make Chronos failed

### ISSUE THREAD

**Issue by anhnhatcs** [autogluon/autogluon#5230]
**Bug Report Checklist**

- [ ] I provided code that demonstrates a minimal reproducible example.  
- [ ] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [x ] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**

**Expected behavior**

**To Reproduce**

**Screenshots / Logs**

Training timeseries model ChronosZeroShot[bolt_base]. 
	Warning: Exception caused ChronosZeroShot[bolt_base] to fail during training... Skipping this model.
	module 'pyarrow' has no attribute 'PyExtensionType'
Training timeseries model ChronosFineTuned[bolt_small]. 
	Warning: Exception caused ChronosFineTuned[bolt_small] to fail during training... Skipping this model.
	module 'pyarrow' has no attribute 'PyExtensionType'

**Installed Versions**

<details>

```python
# Replace this code with the output of the following:
from autogluon.core.utils import show_versions
show_versions()
```
INSTALLED VERSIONS
------------------
date                   : 2025-07-19
time                   : 13:40:46.752440
python                 : 3.10.12.final.0
OS                     : Linux
OS-release             : 5.15.0-1091-azure
Version                : #100-Ubuntu SMP Tue May 27 21:41:06 UTC 2025
machine                : x86_64
processor              : x86_64
num_cores              : 16
cpu_ram_mb             : 50824.0
cuda version           : None
num_gpus               : 0
gpu_ram_mb             : []
avail_disk_size_mb     : 225468

accelerate             : 0.34.2
autogluon              : 1.3.1
autogluon.common       : 1.3.1
autogluon.core         : 1.3.1
autogluon.features     : 1.3.1
autogluon.multimodal   : 1.3.1
autogluon.tabular      : 1.3.1
autogluon.timeseries   : 1.3.1
boto3                  : 1.24.28
catboost               : 1.2.8
coreforecast           : 0.0.12
defusedxml             : 0.7.1
einops                 : 0.8.1
evaluate               : 0.4.0
fastai                 : 2.8.2
fugue                  : 0.9.1
gluonts                : 0.16.2
huggingface-hub        : 0.33.4
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.6
joblib                 : 1.2.0
jsonschema             : 4.23.0
lightgbm               : 4.5.0
lightning              : 2.5.2
matplotlib             : 3.10.3
mlforecast             : 0.13.4
networkx               : 3.4.2
nlpaug                 : 1.1.11
nltk                   : 3.7
numpy                  : 1.26.4
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.3.0
onnx                   : None
onnxruntime            : None
onnxruntime-gpu        : None
openmim                : 0.3.9
optimum                : None
optimum-intel          : None
orjson                 : 3.11.0
pandas                 : 2.2.3
pdf2image              : 1.17.0
Pillow                 : 11.3.0
psutil                 : 5.9.0
pyarrow                : 21.0.0
pytesseract            : 0.3.10
pytorch-lightning      : 2.5.2
pytorch-metric-learning: 2.8.1
ray                    : 2.39.0
requests               : 2.32.4
scikit-image           : 0.25.2
scikit-learn           : 1.5.2
scikit-learn-intelex   : None
scipy                  : 1.15.3
seqeval                : 1.2.2
skl2onnx               : None
spacy                  : 3.5.3
statsforecast          : 1.7.8
tabpfn                 : None
tensorboard            : 2.11.0
text-unidecode         : 1.3
timm                   : 1.0.3
torch                  : 2.2.2
torchmetrics           : 1.7.4
torchvision            : 0.17.2
tqdm                   : 4.64.1
transformers           : 4.49.0
utilsforecast          : 0.2.4
xgboost                : 3.0.2

</details>

**Comment by anhnhatcs** [comment]
I used it as a temporary fixed: 
!pip uninstall pyarrow
!pip install "pyarrow<21.0.0"

**Comment by shchur** [comment]
Hi @anhnhatcs, thanks a lot for reporting the problem. Unfortunately, I could not reproduce it on my side using the same package versions. My output of `show_versions()`:
```
accelerate             : 0.34.2
autogluon              : 1.3.1
autogluon.common       : 1.3.1
autogluon.core         : 1.3.1
autogluon.eda          : 1.3.2b20250723
autogluon.features     : 1.3.1
autogluon.multimodal   : 1.3.1
autogluon.tabular      : 1.3.1
autogluon.timeseries   : 1.3.1
boto3                  : 1.24.28
catboost               : 1.2.8
coreforecast           : 0.0.12
defusedxml             : 0.7.1
einops                 : 0.8.1
evaluate               : 0.4.0
fastai                 : 2.8.2
fugue                  : 0.9.1
gluonts                : 0.16.2
huggingface-hub        : 0.33.4
hyperopt               : 0.2.7
imodels                : 2.0.0
ipython                : 8.12.3
ipywidgets             : 8.1.7
jinja2                 : 3.1.6
joblib                 : 1.2.0
jsonschema             : 4.23.0
lightgbm               : 4.5.0
lightning              : 2.5.2
matplotlib             : 3.10.3
missingno              : 0.5.2
mlforecast             : 0.13.6
networkx               : 3.4.2
nlpaug                 : 1.1.11
nltk                   : 3.7
numpy                  : 1.26.4
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.3.0
onnx                   : 1.17.0
onnxruntime            : 1.19.2
onnxruntime-gpu        : 1.19.2
openmim                : 0.3.9
optimum                : None
optimum-intel          : None
orjson                 : 3.11.0
pandas                 : 2.2.3
pdf2image              : 1.17.0
phik                   : 0.12.4
Pillow                 : 11.3.0
psutil                 : 5.9.0
pyarrow                : 21.0.0
pyod                   : 1.1.3
pytesseract            : 0.3.10
pytorch-lightning      : 2.5.2
pytorch-metric-learning: 2.8.1
ray                    : 2.39.0
requests               : 2.32.4
scikit-image           : 0.25.2
scikit-learn           : 1.5.2
scikit-learn-intelex   : None
scipy                  : 1.15.3
seaborn                : 0.12.2
seqeval                : 1.2.2
shap                   : 0.46.0
skl2onnx               : 1.17.0
spacy                  : 3.5.3
statsforecast          : 1.7.8
suod                   : 0.0.9
tabpfn                 : 2.1.0
tensorboard            : 2.11.0
text-unidecode         : 1.3
timm                   : 1.0.3
torch                  : 2.2.2
torchmetrics           : 1.7.3
torchvision            : 0.17.2
tqdm                   : 4.64.1
transformers           : 4.49.0
utilsforecast          : 0.2.4
xgboost                : 3.0.2
yellowbrick            : 1.5
```
My torch, transformers, pyarrow and numpy versions seem to match yours, but I am able to use Chronos in both zero-shot and fine-tuned mode with:
```python
from autogluon.timeseries import TimeSeriesPredictor

train_url = "https://autogluon.s3.amazonaws.com/datasets/timeseries/m4_hourly/train.csv"
predictor = TimeSeriesPredictor(prediction_length=24)
predictor.fit(
    train_url,
    hyperparameters={
        "Chronos": [
            {},
            {"fine_tune": True, "fine_tune_steps": 20}
        ]
    }
)
```

Do you have any ideas regarding the source of the discrepancy? Are you using some special managed python environment like Databricks?

**Comment by shchur** [comment]
It's possible that this problem is unrelated to `autogluon`, but is rather caused by an old version of `datasets` in your environment (https://github.com/apache/arrow/issues/47155). Maybe when the Chronos model imports `transformers`, this also attempts to import the `datasets` library, which results in this exception. Please let me know if upgrading / removing `datasets` solves this issue.

**Comment by jomach** [comment]
same error here running the examples usecases from start guide

**Comment by shchur** [comment]
I just verified that problem is caused by an old version of `datasets` being installed by default with `pip install autogluon==1.3.1`. 

Current workaround: install newer version of `datasets` with
```
pip install autogluon "datasets>=2.15"
```
or just `pip install datasets>=2.15` if you already installed `autogluon` before.

This problem will be resolved automatically in the upcoming v1.4.0 release of AutoGluon thanks to the pyarrow version cap in #5150.
